rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -187,9 +191,9 @@ void RSocketStateMachine::close(
   VLOG(6) << "close";
 
   if (resumeCallback_) {
-    resumeCallback_->onResumeError(
+    auto resumeCallback = std::move(resumeCallback_);
```

Comment:
ditto

Reply:
Can we do this move at the if check, so if the value is already moved then we will not need to do any check?
```
if (auto resumeCallback = std::move(resumeCallback_) {
   resumeCallback->onResumeError(...);
}
```